lib/sycamore/sycamore/connectors/elasticsearch/elasticsearch_writer.py

```diff
@@ -0,0 +1,140 @@
+            "properties": {
+                "embeddings": {
+                    "type": "dense_vector",
+                    "dims": 384,
+                    "index": True,
+                    "similarity": "cosine",
+                },
+                "properties": {"type": "object"},
+            }
+        }
+    )
+    flatten_properties: bool = False
+
+    def compatible_with(self, other: BaseDBWriter.TargetParams) -> bool:
+        if not isinstance(other, ElasticTargetParams):
+            return False
+        if self.index_name != other.index_name:
+            return False
+        if self.mappings != other.mappings:
+            return False
+        return True
+
+
+class ElasticClient(BaseDBWriter.Client):
+    def __init__(self, client: Elasticsearch):
+        self._client = client
+
+    @classmethod
+    def from_client_params(cls, params: BaseDBWriter.ClientParams) -> "ElasticClient":
+        assert isinstance(params, ElasticClientParams)
+        client = Elasticsearch(params.url, **params.es_client_args)
+        return ElasticClient(client)
+
+    def write_many_records(self, records: list[BaseDBWriter.Record], target_params: BaseDBWriter.TargetParams):
+        assert isinstance(target_params, ElasticTargetParams)
+        assert _narrow_list_of_doc_records(records), f"Found a bad record in {records}"
+        with self._client:
+
+            def insert_into_index():
+                for r in records:
+                    yield {
+                        "_index": target_params.index_name,
+                        "_id": r.doc_id,
+                        "properties": r.properties,
+                        "embeddings": r.embeddings,
+                    }
+
+            for success, info in parallel_bulk(self._client, insert_into_index()):  # generator must be consumed
+                if not success:
+                    print(f"Insert Operation Unsuccessful: {info}")
+
+    def create_target_idempotent(self, target_params: BaseDBWriter.TargetParams):
+        assert isinstance(target_params, ElasticTargetParams)
+        try:
+            self._client.indices.create(index=target_params.index_name, mappings=target_params.mappings, timeout="30s")
+        except ApiError as e:
+            if e.status_code == 400:
+                return
+            raise e
+
+    def get_existing_target_params(self, target_params: BaseDBWriter.TargetParams) -> "ElasticTargetParams":
+        assert isinstance(target_params, ElasticTargetParams)
+        try:
+            data = self._client.indices.get_mapping(index=target_params.index_name)
+            mappings = data[target_params.index_name]["mappings"]
+            if "properties" in mappings and "embeddings" in mappings["properties"]:
+                try:
+                    mappings["properties"]["embeddings"].pop("index_options")
+                except Exception as e:
+                    print(f"Failed to pop index_options from embeddings: {e}")
```

Comment:
generally prefer keeping lots of information in the retrieved target params and handling the inconsistencies from this in `compatible_with`. That's why it's `compatible_with` and not `__eq__`